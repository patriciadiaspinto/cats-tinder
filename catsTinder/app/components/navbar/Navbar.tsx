/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import {Platform, StyleSheet, TouchableOpacity, View} from 'react-native';
import {Message, Paw, User} from '../icons';

const Navbar = () => {
  return (
    <View style={[styles.navbar, styles.shadow]} testID={'navbar-container'}>
      <TouchableOpacity onPress={() => console.log('navigate home')}>
        <Paw />
      </TouchableOpacity>
      <TouchableOpacity onPress={() => console.log('navigate messages')}>
        <Message />
      </TouchableOpacity>
      <TouchableOpacity onPress={() => console.log('navigate profile')}>
        <User />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  navbar: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    gap: 24,
    width: 156,
    borderRadius: 25,
    backgroundColor: 'white',
    paddingVertical: 12,
    paddingHorizontal: 24,
  },
  shadow: {
    shadowColor: '#52006A',
    ...Platform.select({
      ios: {
        shadowOffset: {width: -2, height: 4},
        shadowOpacity: 0.3,
        shadowRadius: 2,
      },
      android: {
        elevation: 20,
      },
    }),
  },
});

export default Navbar;
