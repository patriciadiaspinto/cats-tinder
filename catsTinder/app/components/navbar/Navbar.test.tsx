import {render} from '@testing-library/react-native';
import Navbar from './Navbar';

describe('Navbar component', () => {
  it('should render Navbar', async () => {
    const {getByTestId} = render(<Navbar />);

    expect(getByTestId('navbar-container')).toBeDefined;

    const tree = render(<Navbar />);

    expect(tree).toMatchSnapshot();
  });
});
