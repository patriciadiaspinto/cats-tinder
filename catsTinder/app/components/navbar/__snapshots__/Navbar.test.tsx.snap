// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render Navbar 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 25,
        "display": "flex",
        "flexDirection": "row",
        "gap": 24,
        "justifyContent": "center",
        "paddingHorizontal": 24,
        "paddingVertical": 12,
        "width": 156,
      },
      {
        "shadowColor": "#52006A",
        "shadowOffset": {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 2,
      },
    ]
  }
  testID="navbar-container"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      bbHeight="20"
      bbWidth="20"
      fill="none"
      focusable={false}
      height={20}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M10.746 10.842c2.93.96 6.267 4.681 3.387 6.998-1.466 1.18-2.597.152-4.015-.427-1.736-.709-2.719.468-4.437.275-1.145-.129-2.181-.567-2.656-1.639-1.367-3.094 3.87-5.403 6.366-5.438a4.22 4.22 0 0 1 1.355.231ZM15.67 12.264c.81.186 1.722.056 2.36-.477.639-.533.926-1.501.554-2.245-.635-1.27-2.1-.817-2.87-.007-.364.38-.678.843-.733 1.367-.056.523.689 1.362.689 1.362ZM15.808 5.545c.138-.691.206-1.43-.123-2.08-.33-.65-1.17-1.165-1.965-.994-.692.149-1.111.736-1.4 1.294-.289.56-.52 1.14-.69 1.735-.186.647-.293 1.353.024 1.97 1.357 2.635 3.89-.603 4.154-1.925ZM9.646 7.133c.865-1.08 1.179-2.461.838-3.693-.18-.657-.574-1.3-1.257-1.62C6.327.473 5.763 6.23 7.404 7.33c.718.481 2.242-.196 2.242-.196ZM4.499 8.683c.709-.978.354-2.925-.368-3.813-.31-.383-.757-.708-1.25-.686-.691.03-1.168.71-1.404 1.36C.985 6.897 1.2 9.08 3.03 9.257c.703.068 1.176-.17 1.469-.574Z"
          fill={
            {
              "payload": 4293677950,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      bbHeight="20"
      bbWidth="20"
      fill="none"
      focusable={false}
      height={20}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M17.5 9.583a6.983 6.983 0 0 1-.75 3.167 7.084 7.084 0 0 1-6.333 3.917 6.983 6.983 0 0 1-3.167-.75L2.5 17.5l1.583-4.75a6.983 6.983 0 0 1-.75-3.167A7.083 7.083 0 0 1 7.25 3.25a6.983 6.983 0 0 1 3.167-.75h.416A7.066 7.066 0 0 1 17.5 9.167v.416Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282597697,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <RNSVGSvgView
      bbHeight="20"
      bbWidth="20"
      fill="none"
      focusable={false}
      height={20}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M16.667 17.5v-1.667a3.333 3.333 0 0 0-3.334-3.333H6.667a3.333 3.333 0 0 0-3.334 3.333V17.5M10 9.167A3.333 3.333 0 1 0 10 2.5a3.333 3.333 0 0 0 0 6.667Z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4282597697,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth="2"
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;
